test/test.js

```diff
@@ -179,8 +211,9 @@ describe('options', function() {
       var dir = tree.getPath('home');
       var file = tree.getPath('home/bb/file1');
       watcher = watch(dir, { recursive: true }, function(evt, name) {
-        assert.equal(file, name);
-        done();
+        if (file === name) {
```

Comment:
I'll leave this out in the final commit.

It's here for my Work-In-Progress because I've disabled the native recursive mode for macOS (in has-native-recursive.js), which has exposed a bug in Node.js specific to macOS.

The bug is that an event for a seemingly unrelated file is being emitted by fs.FSWatcher, causing this test to fail. For the disappointing evidence, see <https://github.com/nodejs/node/issues/25290>.